mender.go

```diff
@@ -15,28 +15,30 @@ package main
 
 import (
 	"bufio"
-	"encoding/json"
-	"errors"
-	"io/ioutil"
 	"os"
 	"strings"
 	"time"
 
 	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
 )
 
 type Controler interface {
-	Bootstrap() error
-	TransitionState() MenderState
+	Bootstrap() menderError
 	GetCurrentImageID() string
-	GetDaemonConfig() daemonConfig
-	GetUpdaterConfig() httpsClientConfig
-	LastError() error
+	GetUpdatePollInterval() time.Duration
+	HasUpgrade() (bool, menderError)
+	CheckUpdate() (*UpdateResponse, menderError)
+
+	UInstallCommitRebooter
+	Updater
+	StateRunner
 }
 
 const (
 	defaultManifestFile = "/etc/build_mender"
-	defaultKeyFile      = "/data/mender/mender-agent.pem"
+	defaultKeyFile      = "mender-agent.pem"
+	defaultDataStore    = "/var/lib/mender"
```

Comment:
For now we have agreed on some common location in this ticket https://tracker.mender.io/browse/MEN-441
Ideally one can think of having this (or `$(prefix)` at least) configurable at built time, similarly to what we do with version information.